spark/src/main/resources/python/zeppelin_pyspark.py

```diff
@@ -98,6 +98,10 @@ def reset(self):
         continue
 
       if s[0] != " " and s[0] != "\t":
+        if s.replace(" ", "").startswith("else:") or s.replace(" ", "").startswith("elif") or s.replace(" ", "").startswith("except"):
```

Comment:
Whitespace is significant for Python, perhaps we shouldn't trim it out at all.

it could be a function definition, a if-else, or anything else. It could also be nested too:

```
>>> if foo:
...     if bar:
...         x = 42
... else:
...   print foo
```